### PR TITLE
Split simple partial test classes

### PR DESCRIPTION
**Pull request by roji**
This splits all our "simple" tests, which are currently in a single class (split across files via partial), into separate classes.

Still TODO: set up the Northwind database exactly once for all suites. xunit supports ["shared context"](https://xunit.net/docs/shared-context) via the concept of test collections, but this also [has an impact on parallelism](https://xunit.net/docs/running-tests-in-parallel.html#parallelism-in-test-frameworks). I've posted [a question on stackoverflow](https://stackoverflow.com/questions/58850629/using-shared-context-in-nunit-without-reducing-parallelism), @AndriySvyryd @maumar any ideas on this?

### CODE REVIEW

**Comment by smitpatel** [comment]
- Based on benchmarking I did a while ago, there is not much need to run them in parallel time-wise though it would be good to have.
- Given we already have different test classes using NorthwindFixture which are running in parallel, according to @AndriySvyryd infra should already be capable of running them in parallel without anything specific to be done.

File: test/EFCore.Cosmos.FunctionalTests/Query/NorthwindMiscCosmosTest.cs [hunk 1]
```diff
@@ -14,9 +14,9 @@
 
 namespace Microsoft.EntityFrameworkCore.Query
 {
-    public partial class SimpleQueryCosmosTest : SimpleQueryTestBase<NorthwindQueryCosmosFixture<NoopModelCustomizer>>
+    public class NorthwindMiscCosmosTest : NorthwindMiscTestBase<NorthwindQueryCosmosFixture<NoopModelCustomizer>>
```
Comment (smitpatel) [inline test/EFCore.Cosmos.FunctionalTests/Query/NorthwindMiscCosmosTest.cs]: Prefer not to put short forms like misc in the middle of names.

Comment (smitpatel) [reply test/EFCore.Cosmos.FunctionalTests/Query/NorthwindMiscCosmosTest.cs]: And please some better name. May be SimpleQuery :trollface:

File: test/EFCore.Specification.Tests/Query/KeylessEntitiesTestBase.cs [hunk 2]
```diff
@@ -5,13 +5,26 @@
...
 using System.Threading.Tasks;
 using Microsoft.EntityFrameworkCore.TestModels.Northwind;
+using Microsoft.EntityFrameworkCore.TestUtilities;
 using Xunit;
 
 // ReSharper disable InconsistentNaming
 namespace Microsoft.EntityFrameworkCore.Query
 {
-    public abstract partial class SimpleQueryTestBase<TFixture>
+    public abstract class KeylessEntitiesTestBase<TFixture> : QueryTestBase<TFixture>
```
Comment (smitpatel) [inline test/EFCore.Specification.Tests/Query/KeylessEntitiesTestBase.cs]: I believe the convention is to end with `QueryTestBase` when tests are about query pipeline.

Comment (smitpatel) [reply test/EFCore.Specification.Tests/Query/KeylessEntitiesTestBase.cs]: So `KeylessEntitiesQueryTestBase`

File: test/EFCore.Cosmos.FunctionalTests/Query/NorthwindMiscCosmosTest.cs (hunk 1 above)
Comment (roji) [reply test/EFCore.Cosmos.FunctionalTests/Query/NorthwindMiscCosmosTest.cs]: And so, NorthwindMiscellaneousQueryCosmosTest is born

File: test/EFCore.Specification.Tests/Query/NorthwindAggregateOperatorsQueryTestBase.cs [hunk 3]
```diff
@@ -8,33 +8,26 @@
...
+using Microsoft.EntityFrameworkCore.TestUtilities;
 using Xunit;
 
 #pragma warning disable RCS1202 // Avoid NullReferenceException.
 
 // ReSharper disable InconsistentNaming
 namespace Microsoft.EntityFrameworkCore.Query
 {
-    public abstract partial class SimpleQueryTestBase<TFixture>
+    public abstract class NorthwindAggregateOperatorsQueryTestBase<TFixture> : QueryTestBase<TFixture>
```
Comment (AndriySvyryd) [inline test/EFCore.Specification.Tests/Query/NorthwindAggregateOperatorsQueryTestBase.cs]: Why are they prefixed with Northwind? Usually the test classes are named after the feature they are testing.

Comment (AndriySvyryd) [reply test/EFCore.Specification.Tests/Query/NorthwindAggregateOperatorsQueryTestBase.cs]: Adding Query in the name is also redundant in most cases

---
In reply to: [346674959](https://github.com/aspnet/EntityFrameworkCore/pull/18896#discussion_r346674959) [](ancestors = 346674959)

Comment (smitpatel) [reply test/EFCore.Specification.Tests/Query/NorthwindAggregateOperatorsQueryTestBase.cs]: GearsOfWarQueryTestbase 🗡 We need to determine a consistent convention to name test classes.

**Pull request merged**